src/translations/maps/queuedPhysicalIndexToValueMap.js

```diff
@@ -0,0 +1,103 @@
+import IndexMap from './indexMap';
+import { getListWithRemovedItems, getListWithInsertedItems } from './utils/physicallyIndexed';
+import { getDecreasedIndexes, getIncreasedIndexes } from './utils/actionsOnIndexes';
+import { isFunction } from '../../helpers/function';
+
+/**
+ * Map for storing mappings from an physical index to a value. Some values are stored in a certain order.
+ *
+ * Does not update stored values on remove/add row or column action. Otherwise, queue of indexes related to ordered
+ * values is updated after such changes.
+ */
+class QueuedPhysicalIndexToValueMap extends IndexMap {
+  constructor() {
+    super();
+    this.queueOfIndexes = [];
+  }
+
+  /**
+   * Add values to list and reorganize. It updates queue of indexes related to ordered values.
+   *
+   * @private
+   * @param {number} insertionIndex Position inside the list.
+   * @param {Array} insertedIndexes List of inserted indexes.
+   */
+  insert(insertionIndex, insertedIndexes) {
+    this.indexedValues = getListWithInsertedItems(
+      this.indexedValues,
+      insertionIndex,
+      insertedIndexes,
+      this.initValueOrFn
+    );
+    this.queueOfIndexes = getIncreasedIndexes(this.queueOfIndexes, insertedIndexes);
+
+    super.insert(insertionIndex, insertedIndexes);
+  }
+
+  /**
+   * Remove values from the list and reorganize. It updates queue of indexes related to ordered values.
+   *
+   * @private
+   * @param {Array} removedIndexes List of removed indexes.
+   */
+  remove(removedIndexes) {
+    this.indexedValues = getListWithRemovedItems(this.indexedValues, removedIndexes);
+    this.queueOfIndexes = getDecreasedIndexes(this.queueOfIndexes, removedIndexes);
+
+    super.remove(removedIndexes);
+  }
+
+  /**
+   * Set value at index and add it to the queue of values. Non-default values are stored in a certain order.
+   *
+   * Note: Queued value will be added at the end of the queue.
+   *
+   * @param {number} index The index.
+   * @param {*} value The value to save.
+   */
+  setValueAtIndex(index, value) {
+    super.setValueAtIndex(index, value);
+
+    this.queueOfIndexes.push(index);
+  }
+
+  /**
+   * Remove every queued value.
+   *
+   * Note: Please keep in mind that clear for the rest of the map won't be executed.
+   */
+  clear() {
+    if (isFunction(this.initValueOrFn)) {
+      this.queueOfIndexes.forEach((physicalIndex) => {
+        super.setValueAtIndex(physicalIndex, this.initValueOrFn(physicalIndex));
+      });
+
+    } else {
+      this.queueOfIndexes.forEach((physicalIndex) => {
+        super.setValueAtIndex(physicalIndex, this.initValueOrFn);
+      });
+    }
+
+    this.queueOfIndexes = [];
+  }
+
+  /**
+   * Get every queued value.
```

Comment:
```suggestion
   * Get every value in the same order as they were added to the index mapper. 
```

Reply:
I will change it if [the decision](https://github.com/handsontable/handsontable/pull/7276#discussion_r494836980) will be taken.

```suggestion
   * Get every non-default value in the same order as they were added to the index mapper. 
```